File: carrental-fronted/src/service/carrental.service.js
import axios from "axios";

const API_URL= "http://localhost:8080/api";

class CarRentalService{

    getAllCars(){
        return axios.get(API_URL+"/cars")
    }

    getCarById(id){
        return axios.get(API_URL+"/cars/"+id);
    }

    addCar(car){
        return axios.post(API_URL+"/cars",car);
    }

    updateCar(id,car){
        return axios.put(API_URL+"/cars/"+id,car);
    }

    deleteCar(id){
        return axios.delete(API_URL+"/cars/"+id);
    }
}

export default new CarRentalService();

File: Car-Rental-Backend/src/main/java/com/becoder/carrentalbackend/service/CarRentalService.java
package com.becoder.carrentalbackend.service;

import com.becoder.carrentalbackend.model.Booking;
import com.becoder.carrentalbackend.model.Car;

import java.util.List;

public interface CarRentalService {

    List<Car> getAllCars();
    Car getCarById(Long id);
    Car addCar(Car car);
    Car updateCar(Long id, Car car);
    void deleteCar(Long id);

    List<Booking> getAllBookings();
    Booking getBookingById(Long id);
    Booking createBooking(Booking booking);
    Booking updateBooking(Long id, Booking booking);
    void deleteBooking(Long id);
}
